src/Widget/FormBuilder.php

```diff
@@ -0,0 +1,558 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form\Widget;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Yiisoft\Factory\Exceptions\InvalidConfigException;
+use Yiisoft\Form\FormInterface;
+use Yiisoft\Form\Widget\Event\AfterFormRender;
+use Yiisoft\Form\Widget\Event\BeforeFormRender;
+use Yiisoft\Html\Html;
+use Yiisoft\Http\Method;
+
+final class FormBuilder extends Widget
+{
+    public const VALIDATION_STATE_ON_CONTAINER = 'container';
+    public const VALIDATION_STATE_ON_INPUT = 'input';
+    private string $action = '';
+    private string $method = Method::POST;
+    private bool $encodeErrorSummary = true;
+    private string $errorCssClass = 'has-error';
+    private string $errorSummaryCssClass = 'error-summary';
+    private string $inputCssClass = 'form-control';
+    private string $requiredCssClass = 'required';
+    private string $successCssClass = 'has-success';
+    private string $validatingCssClass = 'validating';
+    private string $validationStateOn = self::VALIDATION_STATE_ON_CONTAINER;
+    private bool $enableClientValidation = true;
```

Comment:
I don't think we should port it directly.

Reply:
Done.